datalad/local/foreach.py

```diff
@@ -0,0 +1,275 @@
+            action="store_true",
+            doc="""if given, must be a boolean flag indicating whether
+            the `cmd` is a Python expression to be evaluated, instead of
+            an external command to be executed"""),
+        # Following options are taken from subdatasets
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to operate on.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the input and/or the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        # TODO: assume True since in 99% of use cases we just want to operate
+        #  on present subdatasets?  but having it explicit could be good to
+        #  "not miss any".  But `--fulfilled true` is getting on my nerves
+        fulfilled=Parameter(
+            args=("--fulfilled",),
+            doc="""if given, must be a boolean flag indicating whether
+            to report either only locally present or absent datasets.
+            By default subdatasets are reported regardless of their
+            status""",
+            constraints=EnsureBool() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        # does without any outputs handling
+        # Not sure yet if worth supporting here
+        # contains=Parameter(
+        #     args=('--contains',),
+        #     metavar='PATH',
+        #     action='append',
+        #     doc="""limit report to the subdatasets containing the
+        #     given path. If a root path of a subdataset is given the last
+        #     reported dataset will be the subdataset itself.[CMD:  This
+        #     option can be given multiple times CMD][PY:  Can be a list with
+        #     multiple paths PY], in which case datasets will be reported that
+        #     contain any of the given paths.""",
+        #     constraints=EnsureStr() | EnsureNone()),
+        # TODO: --diff  to provide `diff` record so any arbitrary  git reset --hard etc desire could be fulfilled
+        bottomup=Parameter(
+            args=("--bottomup",),
+            action="store_true",
+            doc="""whether to report subdatasets in bottom-up order along
+            each branch in the dataset tree, and not top-down."""),
+        # Extra options
+        # TODO: should we just introduce --lower-recursion-limit aka --mindepth of find?
+        subdatasets_only=Parameter(
+            args=("-s", "--subdatasets-only"),
+            action="store_true",
+            doc="""whether to exclude top level dataset."""),
+        passthrough=Parameter(  # TODO  could be of use for `run` as well
+            args=("-p", "--passthrough"),
+            action="store_true",
+            doc="""For command line commands, pass-through their output to 
+            the screen instead of capturing/returning as part of the result records."""),
```

Comment:
more of `={return,passthrough}`... with a stretch, I could indeed come up with a point to have a `log` for another possible mode.
I am a bit uncertain about `--output` for the name though.  I think a similar (well - identical) option could eventually be added to `run/rerun` (so many times I wanted that but always needs to come up with a shim... uff), and there we have `--output` to point to output file(s). So I would prefer to avoid `--output` here. Could be `--outputs` (since stdout and stderr) but still would be too close/too confusing. May be `--standard-outputs|-o` ?

Reply:
added `--standard-outputs|--s-o`

Reply:
`--output-streams` might be a clearer distinction from the notion of `outputs` as used w/ `run`.

Reply:
sounds better indeed, I will rename to `--output-streams|--o-s` if no other alternatives suggested, thank you

Reply:
done